packages/typespec-test/test/NetworkAnalytics.Management/generated/typespec-ts/src/api/dataProducts/index.ts

```diff
@@ -119,61 +119,63 @@ export async function _createDeserialize(
     throw createRestError(result);
   }
 
-  result = result as DataProductsCreateLogicalResponse;
+  const _result = result as unknown as DataProductsCreateLogicalResponse;
```

Comment:
Can we add this only when the problem exists instead just add the cast for all of them?

Reply:
+1 unless the logic to determine when the problem happens adds too much complexity

Reply:
Sure let me use the sub model logic to identify this.